taskstats/src/nlas.rs

```diff
@@ -0,0 +1,289 @@
+// SPDX-License-Identifier: MIT
+
+//use crate::constants::*;
+use anyhow::Context;
+use byteorder::{ByteOrder, NativeEndian as ne};
+use netlink_packet_utils::{
+    nla::{Nla, NlaBuffer, NlasIterator},
+    parsers::*,
+    traits::*,
+    DecodeError,
+};
+use std::mem::size_of_val;
+use serde::{Serialize, Deserialize};
+
+
+pub const TASKSTATS_CMD_ATTR_PID: u16 = 1;
+pub const TASKSTATS_CMD_ATTR_TGID: u16 = 2;
+pub const TASKSTATS_CMD_ATTR_REGISTER_CPUMASK: u16 = 3;
+pub const TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK: u16 = 4;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TaskStatsCmdAttrs {
+    Pid(u32),
+    TGid(u32),
+    RegisterCPUMask(String),
+    DeRegisterCPUMask(String),
+}
+
+impl Nla for TaskStatsCmdAttrs {
+    fn value_len(&self) -> usize {
+        use TaskStatsCmdAttrs::*;
+        match self {
+            Pid(v) => size_of_val(v),
+            TGid(v) => size_of_val(v),
+            RegisterCPUMask(s) => s.len() + 1,
+            DeRegisterCPUMask(s) => s.len() + 1,
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        use TaskStatsCmdAttrs::*;
+        match self {
+            Pid(_) => TASKSTATS_CMD_ATTR_PID,
+            TGid(_) => TASKSTATS_CMD_ATTR_TGID,
+            RegisterCPUMask(_) => TASKSTATS_CMD_ATTR_REGISTER_CPUMASK,
+            DeRegisterCPUMask(_) => TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK,
+        }
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        use TaskStatsCmdAttrs::*;
+        match self {
+            Pid(v) => ne::write_u32(buffer, *v),
+            TGid(v) => ne::write_u32(buffer, *v),
+            RegisterCPUMask(s) => {
+                buffer[..s.len()].copy_from_slice(s.as_bytes());
```

Comment:
This does not write tailing `\0`. How about use `CString::into_bytes_with_nul()`?

Reply:
Bcs of that there's there's in the next line:

buffer[s.len()] = 0;

I grepped the code and found ethtool/src/header.rs:80 not very convincing in terms of catchyness.